app/src/main/java/com/example/mystore/ui/home/HomeViewModel.kt

```diff
@@ -37,38 +38,33 @@ class HomeViewModel @Inject constructor(
     }
     var filtersCount by mutableIntStateOf(3)
     var filtersModel = FiltersModel(priceStartRange, priceEndRange, rating, selectedCategories)
-    var shoppingCartCount = dataStoreRepository.getShoppingCart().count()
+    val shoppingCartCount
+        get() = dataStoreRepository.getShoppingCart().count()
 
     init {
         getProducts()
         getCategories()
     }
 
     fun filterProducts(newInput: String? = null) {
-        products.apply {
-            clear()
-            addAll(allProducts.filter {
-                it.price in (priceStartRange..priceEndRange) && it.rating <= rating
-            })
-        }
+        products.update { emptyList() }
```

Comment:
You should still be able to use `.value.apply {}`, no need to update it twice to just reset the data in there.